packages/pluginmanager/src/widget.tsx

```diff
@@ -0,0 +1,307 @@
+    const { model, translator } = options;
+    super();
+    this.model = model;
+    this.addClass('jp-pluginmanager');
+
+    this.trans = translator.load('jupyterlab');
+
+    this.addWidget(new Disclaimer(model, this.trans));
+
+    const header = new Header(model, this.trans);
+    this.addWidget(header);
+
+    const availableList = new AvailableList(model, this.trans);
+    this.addWidget(availableList);
+  }
+  readonly model: PluginListModel;
+  protected trans: TranslationBundle;
+}
+
+class AvailableList extends ReactWidget {
+  constructor(
+    protected model: PluginListModel,
+    protected trans: TranslationBundle
+  ) {
+    super();
+    this.addClass('jp-pluginmanager-AvailableList');
+    model.stateChanged.connect(this.update, this);
+  }
+
+  render(): JSX.Element {
+    return (
+      <>
+        {this.model.statusError !== null ? (
+          <ErrorMessage>
+            {`Error querying installed extensions${
+              this.model.statusError ? `: ${this.model.statusError}` : '.'
+            }`}
+          </ErrorMessage>
+        ) : this.model.isLoading ? (
+          <div className="jp-pluginmanager-loader">
+            {this.trans.__('Updating plugin list…')}
+          </div>
+        ) : (
+          <Table<IEntry>
+            blankIndicator={() => {
+              return <div>{this.trans.__('No entries')}</div>;
+            }}
+            sortKey={'plugin-id'}
+            rows={this.model.available
+              .filter(pkg => {
+                const pattern = new RegExp(this.model.query, 'i');
+                return (
+                  pattern.test(pkg.id) ||
+                  pattern.test(pkg.extension) ||
+                  (pkg.tokenLabel && pattern.test(pkg.tokenLabel))
+                );
+              })
+              .map(data => {
+                return {
+                  data: data,
+                  key: data.id
+                };
+              })}
+            columns={[
+              {
+                id: 'plugin-id',
+                label: this.trans.__('Plugin'),
+                renderCell: (row: IEntry) => (
+                  <>
+                    <code>{row.id}</code>
+                    <br />
+                    {row.description}
+                  </>
+                ),
+                sort: (a: IEntry, b: IEntry) => a.id.localeCompare(b.id)
+              },
+              {
+                id: 'description',
+                label: this.trans.__('Description'),
+                renderCell: (row: IEntry) => <>{row.description}</>,
+                sort: (a: IEntry, b: IEntry) =>
+                  a.description.localeCompare(b.description),
+                isHidden: true
+              },
+              {
+                id: 'autostart',
+                label: this.trans.__('Autostart?'),
+                renderCell: (row: IEntry) => (
+                  <>
+                    {row.autoStart ? this.trans.__('Yes') : this.trans.__('No')}
```

Comment:
We could add support for `defer` here.

Reply:
Done in 9aa4edd8503d6da48af50da52f0caa89b0cb4e4c.